src/plugins/crypto.c

```diff
@@ -1204,45 +1215,106 @@ gboolean bd_crypto_luks_add_key (const gchar *device, const gchar *pass, const g
     gsize buf_len = 0;
     gchar *nkey_buf = NULL;
     gsize nbuf_len = 0;
+    struct crypt_device *cd = NULL;
+    gint ret = 0;
+    guint64 progress_id = 0;
+    gchar *msg = NULL;
+
+    msg = g_strdup_printf ("Started adding key to the LUKS device '%s'", device);
+    progress_id = bd_utils_report_started (msg);
+    g_free (msg);
 
     if (!pass && !key_file) {
         g_set_error (error, BD_CRYPTO_ERROR, BD_CRYPTO_ERROR_NO_KEY,
                      "No passphrase nor key file given, cannot add key.");
+        bd_utils_report_finished (progress_id, (*error)->message);
```

Comment:
Just thinking about adding `g_warn_if_fail (error != NULL)` since you now dereference `error`.

Reply:
Lets deal with this in #527